### PR TITLE
Add functionality to generate timetables

### PR DESCRIPTION
**Pull request by williamchau123**
## Overview
Implemented functions to generate timetable from a set of courses

## Functions
checkOverlap: given a timetable, check if there are any course conflict for each day.
createTimetable: create a timetable from a set of sections, check overlap for the timetable created.
generateTimetables: takes in a set of courses and calls combinations functions to generate the sections and create timetables from those sections. 

## Update
updated types file to include  timetable object

## Test cases
1 course, 1 lecture and 2 practices Passed.
2 courses, 1 with 1 lecture and 2 practice, 1 with 1 lecture and 1 practice Passed.

## Issue
Currently, there is no course code and instructor for the timetable sections.
This can be fixed from course api side or can be fixed in the functions.

### CODE REVIEW

File: packages/timetable-planner/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { courseCombinations, courseMeetingSectionCombinations, } from "./combinations/combinations"
+
+const checkOverlapForDay = (timetable: Timetable, day: string) => {
+    let section = 0
+    while (section < timetable[day].length) {
+        let section2 = +section + +1
```
Comment (ArsalaBangash) [inline packages/timetable-planner/src/index.ts:13]: This is a complex one liner. Please explain what's going on.

**Review by ArsalaBangash — CHANGES_REQUESTED** [review]
Please document the functions and then this PR is ready to be merged :)

File: packages/timetable-planner/src/index.ts (hunk 1 above)
Comment (williamchau123) [reply packages/timetable-planner/src/index.ts:13]: I am not sure what do you mean for explaining this line?

**Review by ArsalaBangash — APPROVED** [review]
Great work!

**Pull request merged**